### DISCUSSION TITLE
define the symmetry of the basis

### DISCUSSION THREAD

**Discussion by messicr458** [QuSpin/QuSpin#603] [General]
I want to calculate the bose hubbard model with specific symmetry. For example, the conservation of both charges (i.e. the boson
particle number) and dipole moment (i.e. the boson center of mass), as shown in Arxiv:2210,11072. However, I don't know how to define it.

**Comment by mgbukov** [comment]
I think the dipole moment constrains the available spin configurations in
the basis.

Check out the tutorial on the user_basis on the documentation website. This
should give you an idea.

Marin

**Reply by messicr458** [reply]
Thanks for your reply. I have read the tutorial and still failed to construct the correct basis......

**Comment by messicr458** [comment]
Is it possible for me to choose the basis by their index and reconstruct the Hilbert space?
